### PR TITLE
[Hue] 2.1 Update

### PR DESCRIPTION
**Pull request by pindab0ter**
## Description

* Connectivity
  * Enable setting manual IP address and/or username (API key)
  * Improve how error messages are shown when linking a Bridge
  * Add retry action when no Hue Bridge was found
  * Update Hue Bridge discovery and linking code
  * Remove [`node-hue-api`](https://www.npmjs.com/package/node-hue-api) dependency
* Front end
  * Use coloured squares with icons in them for lights
  * Improve scene/group gradients
  * Improve responsiveness when toggling groups
  * Rate limit group toggle action
* Rework "Turn Off All Lights" into "Toggle All Lights"
  * Turns off all lights if any are on, turns on all lights if all are off.
  * Updates state in background every 10 minutes to help "stale while revalidate".

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by pindab0ter** [comment]
I'm currently on holiday. Please keep this open as I plan to continue working on this branch after my return.

**Comment by michaelwschultz** [comment]
@pindab0ter excited to test this out as the previous release wouldn't link to my v1 Bridge.

**Comment by pernielsentikaer** [comment]
Hi @pindab0ter 👋 

Is this ready? Could you help resolving the merge conflicts too 🙂

**Comment by pindab0ter** [comment]
> Hi @pindab0ter 👋
> 
> Is this ready? Could you help resolving the merge conflicts too 🙂

I didn't know anyone contributed to the extension. How come I wasn't tagged for a code review?

I rebased the branch onto main and merged the conflicts. If possible, I'd like someone to go through the changes. I've tested them locally, but I'd like a second opinion.

**Comment by pernielsentikaer** [comment]
Hi @pindab0ter 👋 

Its because it was done [directly](https://github.com/raycast/extensions/commit/66c73a9e96efecfa13b22337e22cd7c5c6af0701) on the main branch without a PR due to an error in the extension 🙂

**Comment by pindab0ter** [comment]
Fair enough!

File: extensions/hue/group-off.afphoto
Comment (sxn) [inline extensions/hue/group-off.afphoto:1]: I'm guessing this file isn't supposed to be part of the PR? 😺

File: extensions/hue/light-off.afphoto
Comment (sxn) [inline extensions/hue/light-off.afphoto:1]: same with this one, I guess this shouldn't be in the PR? 😺

**Comment by pindab0ter** [comment]
I chose to include those so that the graphics can be adjusted without having to be recreated.

**Comment by sxn** [comment]
alright, let's merge it then :shipit:

**Comment by pindab0ter** [comment]
It looks like there's an issue with creating new connections to the Hue Bridge. I'm working on a solution.

**Comment by pindab0ter** [comment]
Fixed in #7334.

**Pull request merged**